### PR TITLE
Bloom filter to provide approximate set-like operations

### PR DESCRIPTION
**Pull request by smastelini**
One more addition to `river.sketch`!

### CODE REVIEW

File: river/sketch/set.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            In case the hyperparameter values of the two involved `sketch.Set` instances do not match.
+        """
+        return self & other
+
+    def union(self, other: "Set"):
+        """Set union.
+
+        Return a new instance that results from the set union between the current `Set` object and `other`.
+        Dunder operators can be used to replace the method call, i.e., `a |= b` and `a | b` for inplace
+        nd non-inplace unions, respectively.
```
Comment (smastelini) [inline river/sketch/set.py:243]:
```suggestion
        and non-inplace unions, respectively.
```

File: river/sketch/set.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        The maximum capacity of the Bloom filter, i.e., the maximum number of distinct items to store given
+        the selected `fp_rate`.
+    fp_rate
+        The allowed rate of false positives. The probability of obtaining a true positive is `1 - fp_rate`.
+    seed
+        Random seed for reproducibility.
+
+    Examples
+    --------
+    >>> import random
```
Comment (MaxHalford) [inline river/sketch/set.py:43]:
```suggestion

    >>> import random
```

File: river/sketch/set.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    >>> from river import sketch
+
+    >>> rng = random.Random(42)
+    >>> s_set = sketch.Set(capacity=100, seed=0)
+
+    We can retrieve the number of selected hash functions:
```
Comment (MaxHalford) [inline river/sketch/set.py:49]:
```suggestion

    We can retrieve the number of selected hash functions:
```

File: river/sketch/set.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    Notes
+    -----
+    This implementation uses an integer to represent the binary array. Bitwise operations are performed in the
+    integer to reflect the Bloom filter updates.
+
+    References
+    ----------
+    [^1]: [Florian Hartmann's blog article on Bloom Filters](https://florian.github.io/bloom-filters/).
+    [^2]: [Wikipedia entry on Bloom filters](https://en.wikipedia.org/wiki/Bloom_filter).
+    """
```
Comment (MaxHalford) [inline river/sketch/set.py:117]:
```suggestion
    [^2]: [Wikipedia entry on Bloom filters](https://en.wikipedia.org/wiki/Bloom_filter).
    
    """
```

**Review by MaxHalford — APPROVED** [review]
Really really cool! I just made some suggestions regarding spaces. It's more consistent to add spaces around sentences in the docstrings. I also think yamp needs that for parsing.

**Comment by MaxHalford** [comment]
@smastelini can you add a release note?

**Comment by smastelini** [comment]
Oh, my bad! I ended up git checking out the release notes while checking if the docs were built correctly. I'll do that!

**Comment by smastelini** [comment]
Done!

File: river/sketch/set.py [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+    >>> rng = random.Random(42)
+    >>> s_set = sketch.Set(capacity=100, seed=0)
+
+
+    We can retrieve the number of selected hash functions:
+    >>> s_set.n_hash
+    7
+
+    And the size of the binary array used by the Bloom filter:
```
Comment (MaxHalford) [inline river/sketch/set.py:55]: I would add carriage return here too. I didn't comment at every sentence, but you get the idea :). Basically one carriage return between each sentence and each code block

**Review by MaxHalford — APPROVED** [review]

File: river/sketch/set.py (hunk 5 above)
Comment (smastelini) [reply river/sketch/set.py:55]: I see! Should I add one carriage return just after a code block or before too?

**Pull request merged**